### DISCUSSION TITLE
Scripting: intercept request and sending response immediately

### DISCUSSION THREAD

**Discussion by wabiloo** [ProxymanApp/Proxyman#2101] [General]
Hi,

Using a mix of the Reverse Proxy and Scripting, I am effectively using Proxyman as a lightweight web server that can return simple payloads back.

Currently my setup is:
- Reverse proxy on 8001, sends the request to "fake-server.com:443"
- Script on URL "fake-server.com" (and all subpath) intercepts the call
   - onRequest sends it to httpbin.org
   - onResponse gets the response from httpbin.org and effectively overwrites the response (in my case by returning a 307)

Now, the call to httpbin.org is actually totally unnecessary and is making things slower than I need.
Is there a way to improve this so that the onRequest is immediately returning a response, or at least not firing a call to an external service?

```
async function onRequest(context, url, request) {
  console.log(url);

  originalPath = request.path
  
  // redirect to httpbin.org, just so that the call does not fail
  request.scheme = "https"
  request.host = "httpbin.org"
  request.path = "anything" + originalPath 
  
  console.log("redirected to httpbin: " + request.path)
  
  // Done
  return request;
}
```

One advantage of the call to httpbin however is that I can see it in the Proxyman list (and therefore examine it). I would not want to lose the ability to inspect the incoming call.

Thanks in advance!

**Comment by NghiaTranUIT** [comment]
You can tick on the `Use as Mock API` in the Script Editor.

![Screenshot 2024-08-07 at 10 37 22 PM](https://github.com/user-attachments/assets/eae15a0b-77ff-4ab7-a0ac-17b2dacbb0b5)

It will:
- Never make a real request to your server
- It return your Response immediately as a Mock API does

**Reply by wabiloo** [reply]
You're so quick to respond!
I did indeed find that in the documentation, but for me the tick box is always disabled...  Not sure why

And: will the request/response still appear in the main list?

**Reply by wabiloo** [reply]
Seems that to get Mock API to be enabled, I first have to disable the support for multiple chained scripts

**Reply by NghiaTranUIT** [reply]
>  will the request/response still appear in the main list?

Yes, it will display both Request and Response.

> Seems that to get Mock API to be enabled, I first have to disable the support for multiple chained scripts

You're right, please disable it first. The "Multiple Scripts feature" doesn't work with the Mock API.
